src/front/glsl/lex_tests.rs

```diff
@@ -158,3 +158,89 @@ fn glsl_lex_version() {
         "Identifier((TokenMetadata { line: 0, chars: 13..17 }, \"core\"))"
     );
 }
+
+#[test]
+fn glsl_lex_operators() {
+    let source = "+ - * | & % / += -= *= |= &= %= /= ++ -- || && ^^";
+    let lex = Lexer::new(source);
+    let tokens: Vec<Token> = lex.collect();
+    assert_eq!(tokens.len(), 19);
+
+    let mut iter = tokens.iter();
+    assert_eq!(
+        format!("{:?}", iter.next().unwrap()),
+        "Plus(TokenMetadata { line: 0, chars: 0..1 })"
```

Comment:
why are we converting the data to string, just to then compare with the other piece of data? Would it be simpler to compare the data directly, i.e.:
```rust
    assert_eq!(
        iter.next(),
        Some(Plus(TokenMetadata { line: 0, chars: 0..1 })),
  );
```

Reply:
The parser's `Token` doesn't implement `PartialEq`, guessing that could be added, but don't think that could be done only for tests.

Also not sure if output for failing tests would be as nice. Here is how it looks with string:

```
---- front::glsl::lex_tests::glsl_lex_version stdout ----                                                 
thread 'main' panicked at 'assertion failed: `(left == right)`                                            
  left: `"Identifier((TokenMetadata { line: 0, chars: 13..17 }, \"core\"))"`,                             
 right: `"Identifier((TokenMetadata { line: 0, chars: 12..17 }, \"core\"))"`', src/front/glsl/lex_tests.rs:156:5
```

Reply:
Let's try that! To implement it for tests only you could do
```rust
#[cfg_attr(test, derive(PartialEq))]`
```

Reply:
Ok, will give that a try and see how it looks for failures

Reply:
Seems to work well, update pushed.